src/frontends/common/src/plugin_loader.hpp

```diff
@@ -15,6 +15,22 @@ static const char PathSeparator[] = ":";
 namespace ov {
 namespace frontend {
 
+/// \brief Internal data structure holding by each frontend. Includes library handle and extensions.
+class FrontEndSharedData {
+    friend inline void add_extension_to_shared_data(std::shared_ptr<void>& obj,
+                                                    const std::shared_ptr<ov::Extension>& ext);
+    std::shared_ptr<void> m_so;
+    std::vector<std::shared_ptr<ov::Extension>> m_loaded_extensions = {};
+
+public:
+    explicit FrontEndSharedData(const std::shared_ptr<void>& so) : m_so(so) {}
+};
+
+inline void add_extension_to_shared_data(std::shared_ptr<void>& obj, const std::shared_ptr<ov::Extension>& ext) {
+    auto obj_data = std::static_pointer_cast<FrontEndSharedData>(obj);
+    obj_data->m_loaded_extensions.push_back(ext);
```

Comment:
nit: Maybe add `OPENVINO_ASSERT(obj_data)` here stating 'internal error', just if some developer will create some other object instead of FrontEndSharedData by mistake

Reply:
done